Slim/Container.php

```
@@ -263,7 +264,30 @@ public function get($id)
         if (!$this->offsetExists($id)) {
             throw new ContainerValueNotFoundException(sprintf('Identifier "%s" is not defined.', $id));
         }
-        return $this->offsetGet($id);
+        try {
+            return $this->offsetGet($id);
+        } catch (\InvalidArgumentException $exception) {
+            if ($this->exceptionThrownByContainer($exception)) {
```

If `(! $this->exceptionThrownByContainer($exception))` the exception is silently discarded, it should be re-thrown.